scripts/leaderboard.coffee

```diff
@@ -278,22 +283,22 @@ module.exports = (robot) ->
     debut = msg.match[1].toLowerCase()
     debut = debut.replace /^\s+|\s+$/g, ""
     queries = debut.split /\s+/
-    name_verified = ""
-    name_unset = ""
+    name_not_added = ""
+    name_added = ""
     for name in queries
       ScoreField = scorefield()
       Aliases = aliases()
-      if verifyName(name, ScoreField, Aliases)
-        name_verified = name_verified.concat name.concat(" ")
-      else
+      if !verifyName(name, ScoreField, Aliases) && checkUser(name.toLowerCase())
```

Comment:
why `!verifyName` ?

Reply:
verifyName checks if the field `ScoreField` is set for a `name` 
So the guranteed check to add a user would be to belong to the users list and not have the ScoreField field set wrt their name. All other cases should be rejected.

Reply:
shouldnt it be `!verifyName(name, ScoreField, Aliases) && !checkUser(name.toLowerCase())`?

Reply:
`checkUser` returns true if a user exists in slack, beware this is different from the list of users in plusplus. Therefore the when `checkUser` returns `true` it means that the user is a legitimate user and then only they'll be registered to scoreboard. All this is to prevent duplication of users in the scoreboard and hence prevent complains from people on having low scores.

Reply:
Oh okay I get it now. I was confusing between the plusplus users and regular users.